### DISCUSSION TITLE
New feature: Reorder tabs in session

### DISCUSSION THREAD

**Discussion by iago-costa** [zellij-org/zellij#3079] [Ideas]
## Feature: Reorder tabs

1. Using drag and drop
2. By alphabet
3. By set index in CLI

![image](https://github.com/zellij-org/zellij/assets/43588158/c6fd7e43-365e-4ef6-98e2-d906cdeac8db)

**Comment by rsniezek** [comment]
+1

I am lacking the ability to reorder tabs, especially when accidentally closing one, combined with the fact than new tabs are always added at the end, it makes it impossible to recreate a layout I just had.

**Reply by merikan** [reply]
@rsniezek fyi, you can manually reorder tabs since v0.40.0. This feature request is an enhancement of that.

If you want to move a tab (left and right):
* Use `alt+i` and `alt+o` to reorder.
* Or if you want to map them yourself use [`{ MoveTab "Left|Right"; }`](https://zellij.dev/documentation/keybindings-possible-actions#movetab)

**Reply by rsniezek** [reply]
that works great, thank you!
BTW, is there a way to lock the layout without locking the entire UI? I'd like to prevent closing/opening tabs (and panes if needed) but I'd still like to be able to navigate them.

**Reply by merikan** [reply]
@rsniezek no, not that I know of. 
Maybe you can start zellij with `lock` mode (`default_mode "locked"`) and only add those keybindings that you need when in `lock` mode
What I do is to clean all keybindings in `normal` mode and use `tmux` mode to jump to the other modes. In `tmux` mode I also have keybindings for the actions I do all the time.  [config](https://github.com/merikan/.dotfiles/blob/main/config/zellij/config.kdl)
